Healthcare-Hospital-Management-System/Healthcare-Hospital-Management-System/Infrastructure/DrugClient.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System.Text.Json;
+
+namespace HealthcareHospitalManagementSystem.Services
+{
+    public class DrugClient
+    {
+        private readonly HttpClient _httpClient;
```

Comment:
For the future, it is better to use HttpClientFactory, the main advantage is easier and from the box lifetime handling which can help avoid issues which may appear with simple HttpClient
https://learn.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests